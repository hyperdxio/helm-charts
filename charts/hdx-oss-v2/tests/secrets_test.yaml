suite: Test Secrets
templates:
  - secrets.yaml
tests:
  - it: should always render app secrets
    set:
      clickhouse:
        enabled: false
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - equal:
          path: metadata.name
          value: RELEASE-NAME-hdx-oss-v2-app-secrets
      - equal:
          path: type
          value: Opaque
      - isNotEmpty:
          path: data.api-key
          
  - it: should render clickhouse secrets when clickhouse is enabled
    set:
      clickhouse:
        enabled: true
        config:
          users:
            appUserPassword: "test-password"
            otelUserPassword: "test-otel-password"
    asserts:
      - hasDocuments:
          count: 2
      - matchSnapshot: {}
          
  - it: should not render clickhouse secrets when clickhouse is disabled
    set:
      clickhouse:
        enabled: false
    asserts:
      - hasDocuments:
          count: 1