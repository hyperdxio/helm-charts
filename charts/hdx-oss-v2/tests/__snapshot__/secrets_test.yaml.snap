should render clickhouse secrets when clickhouse is enabled:
  1: |
    apiVersion: v1
    data:
      api-key: eHh4eHh4eHgteHh4eC14eHh4LXh4eHgteHh4eHh4eHh4eHh4
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: hdx-oss-v2
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: hdx-oss-v2-0.3.1
      name: RELEASE-NAME-hdx-oss-v2-app-secrets
    type: Opaque
  2: |
    apiVersion: v1
    data:
      appUserPassword: dGVzdC1wYXNzd29yZA==
      otelUserPassword: dGVzdC1vdGVsLXBhc3N3b3Jk
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: hdx-oss-v2
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: hdx-oss-v2-0.3.1
      name: RELEASE-NAME-hdx-oss-v2-clickhouse-secrets
    type: Opaque
